### ISSUE TITLE
Catalog sync fails to sync annotations

### ISSUE THREAD

**Issue by mmisztal1980** [hashicorp/consul-k8s#112]
#### Overview of the Issue

Consul catalog sync fails to sync up service annotations

#### Reproduction Steps
- Install docker for windows and enable kubernetes
- Install consul with helm, set the following params:
  - server.replicas=1
  - server.bootstrapExpect=1
  - catalogSync.enabled="-"

Deploy your pods and services. Annotate your service using consul's annotations:

```
λ kubectl -n demo describe service core-http-api1
Name:              core-http-api1
Namespace:         demo
Labels:            app=core-http-api1
Annotations:       consul.hashicorp.com/service-meta-app=core-http-api1
                   consul.hashicorp.com/service-meta-env=local
                   consul.hashicorp.com/service-port=http
                   consul.hashicorp.com/service-sync=true
                   consul.hashicorp.com/service-tags=local,foo
                   kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"consul.hashicorp.com/service-meta-app":"core-http-api1","consul.hashicorp.com/service-m...
Selector:          app=core-http-api1
Type:              ClusterIP
IP:                10.96.94.3
Port:              http  30080/TCP
TargetPort:        80/TCP
Endpoints:         10.1.14.31:80
Session Affinity:  None
Events:            <none>
```
Port-forward into Consul UI, observe no annotations having been synced:
![image](https://user-images.githubusercontent.com/1357057/58432803-86782480-80b4-11e9-9d27-d1f44b0954eb.png)

![image](https://user-images.githubusercontent.com/1357057/58433211-60538400-80b6-11e9-8341-b0f8d9920e2d.png)

### Consul info for both Client and Server

<details>
  <summary>Client info</summary>

```
λ kubectl -n consul exec -it consul-connect-consul-4jwn9 -- sh
/ # consul info
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 0
        services = 0
build:
        prerelease =
        revision = 34eff659
        version = 1.5.0
consul:
        acl = disabled
        known_servers = 1
        server = false
runtime:
        arch = amd64
        cpu_count = 2
        goroutines = 53
        max_procs = 2
        os = linux
        version = go1.12.1
serf_lan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 3
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 40
        members = 2
        query_queue = 0
        query_time = 1
/ #
```

</details>

<details>
  <summary>Server info</summary>

```
λ kubectl -n consul exec -it consul-connect-consul-server-0 -- sh
/ # consul info
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 0
        services = 0
build:
        prerelease =
        revision = 34eff659
        version = 1.5.0
consul:
        acl = disabled
        bootstrap = true
        known_datacenters = 1
        leader = true
        leader_addr = 10.1.14.35:8300
        server = true
raft:
        applied_index = 7810
        commit_index = 7810
        fsm_pending = 0
        last_contact = 0
        last_log_index = 7810
        last_log_term = 12
        last_snapshot_index = 0
        last_snapshot_term = 0
        latest_configuration = [{Suffrage:Voter ID:f1c12866-2057-73a0-f28a-d2e732f96417 Address:10.1.13.236:8300}]
        latest_configuration_index = 1
        num_peers = 0
        protocol_version = 3
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Leader
        term = 12
runtime:
        arch = amd64
        cpu_count = 2
        goroutines = 95
        max_procs = 2
        os = linux
        version = go1.12.1
serf_lan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 3
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 40
        members = 2
        query_queue = 0
        query_time = 1
serf_wan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 1
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 9
        members = 1
        query_queue = 0
        query_time = 1
/ #
```

</details>

### Operating system and Environment details

Windows 10 Enterprise with Docker for Windows and Kubernetes

### Log Fragments

https://gist.github.com/mmisztal1980/70d18f7488f14cae431f93f8c0bc319d

**Comment by alexpekurovsky** [comment]
Any updates here?

**Comment by lkysow** [comment]
This will be fixed by #141. Sorry for the delay.
